File: HackerRank Solutions/10 Days of Statistics/Day 0 Weighted Mean.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        int [] x = new int[n];
        for(int i=0; i<n; i++) {
            x[i] = scan.nextInt();
        }
        int sum = 0, cnt = 0, w;
        for(int i=0; i<n; i++) {
            w = scan.nextInt();
            sum += x[i] * w;
            cnt += w;
        }
        System.out.printf("%.1f\n", (double)sum/cnt);
    }
}

File: HackerRank Solutions/10 Days of Statistics/Day 0 Weighted Mean/Day 0 Weighted Mean.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
def weightedMean(values:list, weights:list) -> float:
    sumWeightedValues = 0
    sumWeights = 0
    for i in range(len(values)):
        value = values[i]
        weight = weights[i]
        sumWeightedValues += value * weight
        sumWeights += weight
    result = sumWeightedValues / sumWeights
    return result
n = int(input())
x = [int(i) for i in input().split()]
w = [int(i) for i in input().split()]
print(round(weightedMean(x, w), 1))